.vib/rabbitmq/goss/goss.yaml

```
@@ -1,31 +1,46 @@
+addr:
+  tcp://rabbitmq:{{ .Vars.service.ports.amqp }}:
+    reachable: true
+  tcp://rabbitmq:{{ .Vars.service.ports.epmd }}:
+    reachable: true
+  tcp://rabbitmq:{{ .Vars.service.ports.dist }}:
+    reachable: true
+file:
+  {{ .Vars.persistence.mountPath }}:
+    exists: true
+    filetype: symlink
+    mode: "0777"
+    owner: root
+  /bitnami/rabbitmq/conf:
+    exists: true
+    filetype: directory
+    mode: "3777"
+    owner: root
+  /var/run/secrets/kubernetes.io/serviceaccount:
+    exists: {{ .Vars.serviceAccount.automountServiceAccountToken }}
+    filetype: directory
+    mode: "3777"
 command:
-  rabbmitmqctl-version:
-    exec: rabbitmqctl version | grep --quiet $APP_VERSION
-    exit-status: 0
-    stderr: [ ]
-    timeout: 20000
-  rabbitmqctl-add-vhost:
-    exec: rabbitmqctl add_vhost test-host
+  {{- $nodes := .Vars.replicaCount }}
+  rabbitmqctl-cluster-status:
+    exec: rabbitmqctl cluster_status
     exit-status: 0
-    stderr: [ ]
     stdout:
-      - 'Adding vhost'
-    timeout: 20000  
-  rabbitmqctl-list-global-params:
-    exec: rabbitmqctl list_global_parameters
+    {{ range $e, $i := until $nodes }}
+      - /rabbitmq-{{ $i }}.rabbitmq-headless/
+    {{ end }}
+  rabbitmqctl-replication-check:
+    {{- $vhost := printf "vhost_%s" (randAlpha 5) }}
+    exec: rabbitmqctl add_vhost {{ $vhost }} && sleep 2 {{ range $e, $i := until $nodes }} && rabbitmqctl --node rabbit@rabbitmq-{{ $i }} list_vhosts | grep {{ $vhost }} {{ end }}
```

Use the `--quiet` option with `grep` as you are not actively using the output in the tests.

```suggestion
    exec: rabbitmqctl add_vhost {{ $vhost }} && sleep 2 {{ range $e, $i := until $nodes }} && rabbitmqctl --node rabbit@rabbitmq-{{ $i }} list_vhosts | grep -q {{ $vhost }} {{ end }}
```